BannerlordTwitch/BLTAdoptAHero/Actions/AdoptAHero.cs

```diff
@@ -174,6 +178,33 @@ private static (bool success, string message) ExecuteInternal(string args, strin
             {
                 return (false, "You can't adopt a hero: no available hero matching the requirements was found!");
             }
+            
+            if (settings.JoinPlayerCompanion && settings.SpawnInParty)
+            {
+                var mainParty = MobileParty.MainParty;
+                if (mainParty != null && mainParty.Party.NumberOfAllMembers < mainParty.Party.PartySizeLimit)
+                {
+                    AddHeroToPartyAction.Apply(newHero, mainParty);
+                    Log.Info($"Placed new hero {newHero.Name} in hero party");
+                }
+                else
+                {
+                    if(mainParty == null)
+                    {
+                        return (false, "You can't adopt a hero: main hero party don't exist for now!");
+                    }
+                    else
+                    {
+                        return (false, "You can't adopt a hero: main hero party is full!");
+                    }
+                }
+            }
+            else
+            {
+                var targetSettlement = Settlement.All.Where(s => s.IsTown).SelectRandom();
```

Comment:
I missed it before, but this is wrong, the character should ONLY be put into a settlement if they were newly created, which won't be the case unless Create New is checked, so you can make this `else if(settings.CreateNew)`.
As it is currently it will move existing wanderers, and existing nobles into random settlements.